### ISSUE TITLE
Program crashes if I input more than 36 Unicode characters without spaces

### ISSUE THREAD

**Issue by asakura42** [drvog/yat-rs#4]
Oh I know that may be minor issue for cyrillic or same UTF8 langs because there are barely no words that lenght, but some asian languages doesn't use spaces so I think I need to post this issue.

**Comment by lewis-weinberger** [comment]
I can't seem to reproduce this crash, could you provide a few more details please?

If you run `yat` with `RUST_BACKTRACE=1 cargo run --release 2>error.log` you should see what exactly is causing Rust to panic. Could you please post the backtrace at the end of `error.log`? Also what are the dimensions of your terminal (try running `tput cols; tput lines`)?

I suspect this is a problem in the `wrap_print` method of the `Window`. I also noticed I didn't implement anything in the entry dialogue to handle input lines longer than the width of the box, which should be fixed!

**Comment by asakura42** [comment]
@lewis-weinberger 

```
...

thread 'main' panicked at 'byte index 73 is not a char boundary; it is inside 'а' (bytes 72..74) of `аааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааа`', src/tui.rs:170:30
stack backtrace:
   0: rust_begin_unwind
             at /build/rust/src/rustc-1.54.0-src/library/std/src/panicking.rs:515:5
   1: core::panicking::panic_fmt
             at /build/rust/src/rustc-1.54.0-src/library/core/src/panicking.rs:92:14
   2: core::str::slice_error_fail
   3: yat::View::run
   4: yat::main
```

And full backtrace:

```
...

thread 'main' panicked at 'byte index 73 is not a char boundary; it is inside 'а' (bytes 72..74) of `ааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааа`', src/tui.rs:170:30
stack backtrace:
   0:     0x560b13f36ea0 - std::backtrace_rs::backtrace::libunwind::trace::h930cf0be2ce43851
                               at /build/rust/src/rustc-1.54.0-src/library/std/src/../../backtrace/src/backtrace/libunwind.rs:90:5
   1:     0x560b13f36ea0 - std::backtrace_rs::backtrace::trace_unsynchronized::he8c90b95546147bf
                               at /build/rust/src/rustc-1.54.0-src/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x560b13f36ea0 - std::sys_common::backtrace::_print_fmt::h2d518076e7529961
                               at /build/rust/src/rustc-1.54.0-src/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x560b13f36ea0 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hc03afe5108b924f5
                               at /build/rust/src/rustc-1.54.0-src/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x560b13f577fc - core::fmt::write::h03501fd186e1f342
                               at /build/rust/src/rustc-1.54.0-src/library/core/src/fmt/mod.rs:1110:17
   5:     0x560b13f34cb5 - std::io::Write::write_fmt::he6bd3203a99cf841
                               at /build/rust/src/rustc-1.54.0-src/library/std/src/io/mod.rs:1588:15
   6:     0x560b13f38e4b - std::sys_common::backtrace::_print::h6d8d274cdb4b9345
                               at /build/rust/src/rustc-1.54.0-src/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x560b13f38e4b - std::sys_common::backtrace::print::hdbba7107ed5db727
                               at /build/rust/src/rustc-1.54.0-src/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x560b13f38e4b - std::panicking::default_hook::{{closure}}::h69616457d96cfc5a
                               at /build/rust/src/rustc-1.54.0-src/library/std/src/panicking.rs:208:50
   9:     0x560b13f38921 - std::panicking::default_hook::h10bbc125f6ee1853
                               at /build/rust/src/rustc-1.54.0-src/library/std/src/panicking.rs:225:9
  10:     0x560b13f39524 - std::panicking::rust_panic_with_hook::h6eb033573a515ade
                               at /build/rust/src/rustc-1.54.0-src/library/std/src/panicking.rs:622:17
  11:     0x560b13f38ff7 - std::panicking::begin_panic_handler::{{closure}}::hffa4a3da6bf48f5f
                               at /build/rust/src/rustc-1.54.0-src/library/std/src/panicking.rs:519:13
  12:     0x560b13f3739c - std::sys_common::backtrace::__rust_end_short_backtrace::h8245d4af937e663d
                               at /build/rust/src/rustc-1.54.0-src/library/std/src/sys_common/backtrace.rs:141:18
  13:     0x560b13f38f59 - rust_begin_unwind
                               at /build/rust/src/rustc-1.54.0-src/library/std/src/panicking.rs:515:5
  14:     0x560b13ebd5d1 - core::panicking::panic_fmt::h497201afa0408fa6
                               at /build/rust/src/rustc-1.54.0-src/library/core/src/panicking.rs:92:14
  15:     0x560b13ebdd56 - core::str::slice_error_fail::h9984bde041d8e577
  16:     0x560b13ecb2fd - yat::View::run::hf50e40e43b22d6f3
  17:     0x560b13ebe401 - yat::main::h9351de55ac0ec648
  18:     0x560b13ebde73 - std::sys_common::backtrace::__rust_begin_short_backtrace::h05ff9b5680339ccc
  19:     0x560b13ebe559 - std::rt::lang_start::{{closure}}::h824ddf93e5cce054
  20:     0x560b13f39a29 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h23da4485ce9c2fa2
                               at /build/rust/src/rustc-1.54.0-src/library/core/src/ops/function.rs:259:13
  21:     0x560b13f39a29 - std::panicking::try::do_call::h26656c7c8049ab32
                               at /build/rust/src/rustc-1.54.0-src/library/std/src/panicking.rs:401:40
  22:     0x560b13f39a29 - std::panicking::try::hafa31606aae6b61d
                               at /build/rust/src/rustc-1.54.0-src/library/std/src/panicking.rs:365:19
  23:     0x560b13f39a29 - std::panic::catch_unwind::h12fc93c48e24c0bf
                               at /build/rust/src/rustc-1.54.0-src/library/std/src/panic.rs:434:14
  24:     0x560b13f39a29 - std::rt::lang_start_internal::hb67ef9ab47b75f78
                               at /build/rust/src/rustc-1.54.0-src/library/std/src/rt.rs:34:21
  25:     0x560b13ebe492 - main
  26:     0x7f514371ab25 - __libc_start_main
  27:     0x560b13ebdd8e - _start
  28:                0x0 - <unknown>

```

cols x lines = 170x45

**Comment by lewis-weinberger** [comment]
Thanks for the info @asakura42. I think I should have solved this crash in 12b9b08 by fixing the `wrap_print` function for multi-byte wide characters. Let me know if you still have any crashes.

I'm going to keep this issue open because I still need to improve the dialogue box so that it's easier to input lines which are wider than the terminal.

**Comment by lewis-weinberger** [comment]
Hi @asakura42, in 9c10cfb I've updated the dialogue box so that now when you enter input wider than the terminal it will span across multiple lines. I think this is working for multi-byte wide unicode characters but please do raise another issue if you have any crashes or unexpected behaviour.
